src/app/core/filter/filter-generator/filter-generator.service.ts

```diff
@@ -56,11 +57,65 @@ export class FilterGeneratorService {
       const label = filterConfig.label ?? schema.labelShort ?? schema.label;
       const type = filterConfig.type ?? schema.dataType;
       if (type == "configurable-enum") {
+        // Add invalid and empty options
+        const enumValues =
+          this.enumService.getEnumValues(schema.additional) || [];
+        const validIds = new Set(enumValues.map((ev) => ev.id));
+        // Get all unique values from data for this field (by id if object, or value)
+        const dataValues = [
+          ...new Set(
+            (data ?? []).map((e) => {
+              const v = e?.[filterConfig.id];
+              if (v && typeof v === "object" && "id" in v) return v.id;
+              return v;
+            }),
+          ),
+        ];
+
+        // Find invalid options (not in enum)
+        const invalidOptions = dataValues
+          .filter(
+            (v) =>
+              v !== undefined && v !== null && v !== "" && !validIds.has(v),
+          )
+          .map((invalidId) => ({
+            key: "__invalid__:" + invalidId,
+            label: `[Invalid: ${invalidId}]`,
+            color: "#d9534f",
+            isInvalid: true,
+            filter: { [filterConfig.id + ".id"]: invalidId } as DataFilter<T>,
+          }));
+
+        // Add "empty" option if there are empty/undefined values
+        const hasEmpty = dataValues.some(
+          (v) => v === undefined || v === null || v === "",
+        );
+        const emptyOption = hasEmpty
+          ? [
+              {
+                key: "__empty__",
+                label: "not defined",
+                color: "#888",
```

Comment:
we wanted to use the same css class as for the enitty-count-dashboard, to ensure consistent styling (even if it should change in the future).

--> try to
1. create a second scss file and move the `not-defined-label` class (and one for invalid) to that
2. import it both in entity-count-dashboard and this component
3. use an `ngClass` to set the color conditionally for these special entries, instead of adding the color here